ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/objectivefunctionevaluator/RemedialActionCostEvaluator.java

```diff
@@ -0,0 +1,93 @@
+ */
+public class RemedialActionCostEvaluator implements CostEvaluator {
+    private final Set<State> optimizedStates;
+    private final Set<FlowCnec> flowCnecs;
+    private final Unit unit;
+    private final MarginEvaluator marginEvaluator;
+    private final RangeActionsOptimizationParameters rangeActionsOptimizationParameters;
+
+    public RemedialActionCostEvaluator(Set<State> optimizedStates, Set<FlowCnec> flowCnecs, Unit unit, MarginEvaluator marginEvaluator, RangeActionsOptimizationParameters rangeActionsOptimizationParameters) {
+        this.optimizedStates = optimizedStates;
+        this.flowCnecs = flowCnecs;
+        this.unit = unit;
+        this.marginEvaluator = marginEvaluator;
+        this.rangeActionsOptimizationParameters = rangeActionsOptimizationParameters;
+    }
+
+    @Override
+    public String getName() {
+        return "remedial-action-cost-evaluator";
+    }
+
+    @Override
+    public Pair<Double, List<FlowCnec>> computeCostAndLimitingElements(FlowResult flowResult, RemedialActionActivationResult remedialActionActivationResult, Set<String> contingenciesToExclude) {
+        return Pair.of(getTotalNetworkActionsCost(remedialActionActivationResult) + getTotalRangeActionsCost(remedialActionActivationResult), EvaluatorsUtils.getCostlyElements(flowCnecs, marginEvaluator, unit, flowResult, contingenciesToExclude));
+    }
+
+    private double getTotalNetworkActionsCost(RemedialActionActivationResult remedialActionActivationResult) {
+        return remedialActionActivationResult.getActivatedNetworkActions().stream().mapToDouble(networkAction -> networkAction.getActivationCost().orElse(0.0)).sum();
+    }
+
+    private double getTotalRangeActionsCost(RemedialActionActivationResult remedialActionActivationResult) {
+        double totalRangeActionsCost = 0d;
+        // TODO: this should maybe only be used for MILP equations but not in the evaluator
+        for (State state : optimizedStates) {
+            for (RangeAction<?> rangeAction : remedialActionActivationResult.getActivatedRangeActions(state)) {
+                totalRangeActionsCost += rangeAction.getActivationCost().orElse(0d);
+                if (rangeAction instanceof PstRangeAction) {
+                    totalRangeActionsCost += computeVariationCost(rangeAction, rangeActionsOptimizationParameters.getPstPenaltyCost(), state, remedialActionActivationResult);
+                } else if (rangeAction instanceof InjectionRangeAction) {
+                    totalRangeActionsCost += computeVariationCost(rangeAction, rangeActionsOptimizationParameters.getInjectionRaPenaltyCost(), state, remedialActionActivationResult);
+                } else if (rangeAction instanceof HvdcRangeAction) {
+                    totalRangeActionsCost += computeVariationCost(rangeAction, rangeActionsOptimizationParameters.getHvdcPenaltyCost(), state, remedialActionActivationResult);
+                } else {
```

Comment:
This can be dangerous when we're adding a new type of rangeAction. Make an else if statement for each type of rangeAction and throw in the else